routes/filter.go

```diff
@@ -0,0 +1,217 @@
+	var args = &FilterArgs{}
+
+	if err := r.bindQuery(c, args); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"Error": err.Error()})
+		return
+	}
+
+	if args.MaxResult == 0 {
+		args.MaxResult = 15
+	}
+	if args.Page == 0 {
+		args.Page = 1
+	}
+	if args.Sorting == "" {
+		args.Sorting = "-created_at"
+	}
+
+	switch resource {
+	case "project":
+		projects, err := models.ProjectAPI.FilterProjects(models.GetProjectArgs{
+			MaxResult:         args.MaxResult,
+			Page:              args.Page,
+			Sorting:           args.Sorting,
+			FilterID:          args.ID,
+			FilterSlug:        args.Slug,
+			FilterTitle:       args.Title,
+			FilterDescription: args.Description,
+			FilterTagName:     args.Tag,
+			FilterPublishedAt: args.PublishedAt,
+			FilterUpdatedAt:   args.UpdatedAt,
+			Fields:            []string{"project_id", "title", "slug", "progress", "status", "publish_status", "published_at"},
+		})
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"Error": err.Error()})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{"_items": projects})
+	case "post":
+		posts, err := models.PostAPI.FilterPosts(&models.PostArgs{
+			MaxResult:         uint8(args.MaxResult),
+			Page:              uint16(args.Page),
+			Sorting:           args.Sorting,
+			FilterID:          args.ID,
+			FilterTitle:       args.Title,
+			FilterContent:     args.Content,
+			FilterAuthorName:  args.Author,
+			FilterTagName:     args.Tag,
+			FilterPublishedAt: args.PublishedAt,
+			FilterUpdatedAt:   args.UpdatedAt,
+		})
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"Error": err.Error()})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{"_items": posts})
+	case "asset":
+		assets, err := asset.AssetAPI.FilterAssets(&asset.GetAssetArgs{
+			MaxResult:       uint8(args.MaxResult),
+			Page:            uint16(args.Page),
+			Sorting:         args.Sorting,
+			FilterID:        args.ID,
+			FilterTitle:     args.Title,
+			FilterTagName:   args.Tag,
+			FilterCreatedAt: args.CreatedAt,
+			FilterUpdatedAt: args.UpdatedAt,
+		})
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"Error": err.Error()})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{"_items": assets})
+	case "member":
+		members, err := models.MemberAPI.FilterMembers(&models.MemberArgs{
+			MaxResult:       uint8(args.MaxResult),
+			Page:            uint16(args.Page),
+			Sorting:         args.Sorting,
+			FilterID:        args.ID,
+			FilterMail:      args.Mail,
+			FilterNickname:  args.Nickname,
+			FilterCreatedAt: args.CreatedAt,
+			FilterUpdatedAt: args.UpdatedAt,
+		})
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"Error": err.Error()})
+			return
+		}
+		c.JSON(http.StatusOK, gin.H{"_items": members})
+	default:
+		c.JSON(http.StatusBadRequest, gin.H{"Error": "Resource Not Supported"})
+	}
+
+}
+
+func (r *filterHandler) SetRoutes(router *gin.Engine) {
+
+	filterRouter := router.Group("/filter")
```

Comment:
Since members, posts, projects and assets all have corresponding functions in models to parse filter on each fields, and the outcome are still of the same type, it feels like filter is not a new kind of resource, but part of the these APIs.

I feel `/members/filter?updated_at=1234567890` is much more reasonable than `/filter/member?updated_at=1234567890`

Reply:
makes sense, will fix it